sql/hive/src/main/java/org/apache/hadoop/hive/ql/io/orc/VectorizedSparkOrcNewRecordReader.java

```diff
@@ -0,0 +1,320 @@
+      return false;
+    }
+
+    @Override
+    public boolean isNullAt(int ordinal) {
+      ColumnVector col = columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return col.isNull[0];
+      } else {
+        return col.isNull[rowId];
+      }
+    }
+
+    @Override
+    public boolean getBoolean(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return col.vector[0] > 0;
+      } else {
+        return col.vector[rowId] > 0;
+      }
+    }
+
+    @Override
+    public byte getByte(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (byte)col.vector[0];
+      } else {
+        return (byte)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public short getShort(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (short)col.vector[0];
+      } else {
+        return (short)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public int getInt(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (int)col.vector[0];
+      } else {
+        return (int)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public long getLong(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (long)col.vector[0];
+      } else {
+        return (long)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public float getFloat(int ordinal) {
+      DoubleColumnVector col = (DoubleColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (float)col.vector[0];
+      } else {
+        return (float)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public double getDouble(int ordinal) {
+      DoubleColumnVector col = (DoubleColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (double)col.vector[0];
+      } else {
+        return (double)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public Decimal getDecimal(int ordinal, int precision, int scale) {
+      DecimalColumnVector col = (DecimalColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return Decimal.apply(col.vector[0].getHiveDecimal().bigDecimalValue(), precision, scale);
+      } else {
+        return Decimal.apply(col.vector[rowId].getHiveDecimal().bigDecimalValue(),
+          precision, scale);
+      }
+    }
+
+    @Override
+    public UTF8String getUTF8String(int ordinal) {
+      BytesColumnVector bv = ((BytesColumnVector)columns[columnIDs.get(ordinal)]);
+      String str = null;
+      if (bv.isRepeating) {
+        str = new String(bv.vector[0], bv.start[0], bv.length[0], StandardCharsets.UTF_8);
```

Comment:
Can creation of a String be avoided by using `UTF8String.fromBytes`? My understanding is that the encode/decode in `new String(..)` and `UTF8String.fromString` can add up.